### PR TITLE
Online booking tweaks

### PR DESCRIPTION
**Pull request by benbarnett**
* Minor copy tweaks
* Only show ineligibility page if all other errors are satisfied

### CODE REVIEW

File: app/controllers/booking_requests_controller.rb [hunk 1]
```diff
@@ -14,12 +14,12 @@ def step_two
...
+    if @booking_request.step_two_valid_excluding_eligibility?
+      return redirect_to booking_request_ineligible_location_path(id: location_id) unless @booking_request.eligible?
+
       BookingRequests.create(@booking_request)
       redirect_to booking_request_completed_location_path(id: location_id)
     else
-      return redirect_to booking_request_ineligible_location_path(id: location_id) unless @booking_request.eligible?
-
       render :step_two
     end
```
Comment (dwhenry) [inline app/controllers/booking_requests_controller.rb]: Don't feel like this change improved readability.

Have you consider:
```
if @booking_request.step_two_valid? && @booking_request.eligible?
  # do A
elsif @booking_request.step_two_valid? # i.e. ineligible
  # do B
else
  # do C
end
```

The reason I suggest this is that step 2 is complete (i.e. valid) regardless of eligibility criteria.

File: app/forms/booking_request_form.rb [hunk 2]
```diff
@@ -53,6 +53,13 @@ def step_two_valid?
     valid?
   end
 
+  def step_two_valid_excluding_eligibility?
+    @step = 2
+    valid?
+
+    (errors.keys - %i(appointment_type dc_pot)).empty?
```
Comment (dwhenry) [inline app/forms/booking_request_form.rb]: if we are going to do this can we extract `%i(appointment_type dc_pot)` into a constant.

Comment (benlovell) [inline app/forms/booking_request_form.rb]: Agreed, obviously replacing the same in `eligible?` too.

File: app/forms/booking_request_form.rb [hunk 3]
```diff
@@ -20,7 +20,7 @@ class BookingRequestForm
     step_two.validates :memorable_word, presence: true
     step_two.validates :appointment_type, inclusion: { in: %w(50-54 55-plus) }
     step_two.validates :accessibility_requirements, inclusion: { in: %w(0 1) }
-    step_two.validates :opt_in, inclusion: { in: %w(0 1) }
```
Comment (benlovell) [inline app/forms/booking_request_form.rb]: What happened here? Doesn't this mean they *must* opt in now?

File: app/views/booking_requests/step_two.html.erb [hunk 4]
```diff
@@ -139,7 +139,7 @@
           <% if @booking_request.errors.include?(:opt_in) %>
             <span class="error-message"><%= @booking_request.errors[:opt_in].to_sentence %></span>
           <% end %>
-          <%= f.check_box :opt_in, class: 't-opt-in' %> Yes, I would like to receive marketing information
+          <%= f.check_box :opt_in, class: 't-opt-in' %> I accept the <a href="/privacy" target="_blank">terms and conditions</a>
```
Comment (benlovell) [inline app/views/booking_requests/step_two.html.erb]: Ah I see now. Ignore the previous comment regarding validation.

File: app/forms/booking_request_form.rb (hunk 3 above)
Comment (benlovell) [inline app/forms/booking_request_form.rb]: Edit: Ignore this, I realised the meaning below.

File: app/controllers/booking_requests_controller.rb (hunk 1 above)
Comment (benlovell) [inline app/controllers/booking_requests_controller.rb]: I don't think that improves readability either. Perhaps `eligible?` needs to operate like a separate step, following the existing pattern in that form object? I think that would be cleaner. Also we need specs for these behaviour changes as they're fundamental to this process.

**Comment by benlovell** [comment]
I feel like the change to the meaning of `opt_in` should be documented better in the commit message. It might lend some meaning to the naming of `opt_in` at least.

**Comment by benbarnett** [comment]
As just quickly discussed I think we're going to leave the eligibility as they are (were) for now. Will close this PR and re-raise with the copy amends separately 😅

**Comment by benlovell** [comment]
*Tis better to have loved and lost, than never to have loved at all*

**Pull request closed without merging**